### PR TITLE
Update Rust from 1.68.2 to 1.74.1

### PR DESCRIPTION
**Pull request by quaresmajose**
Populate kirkstone/rust branch (cover letter only)
```
As discussed on the last couple of YP technical calls, I have created
a kirkstone/rust branch in meta-lts-mixins.  It backports the state of
As discussed on the last couple of YP technical calls, I have created
a kirkstone/rust branch in meta-lts-mixins.  It backports the state of
Rust support in master to kirkstone (so currently 1.74.1) the same way
that the previous kirkstone/rust-1.68 and kirkstone/rust-1.70 branches
did against mickledore and nanbield.  The aim with kirkstone/rust is to
keep it up to date with the level of Rust support in master until the
Kirkstone EOL in April 2026.

The majority of the patches listed below were manually picked from
master and adapted on top of the kirkstone/rust-1.70 branch to create
this new branch.  If you would like to look at the changes in detail,
they are browsable at:

https://git.yoctoproject.org/meta-lts-mixins/log/?h=kirkstone/rust
```

https://lists.yoctoproject.org/g/yocto/message/62278

### CODE REVIEW

**Review by igoropaniuk — APPROVED** [review]
LGTM

**Review by angolini — APPROVED** [review]

**Comment by ricardosalveti** [comment]
Please rebase and trigger a test round.

File: setup-environment-internal [hunk 1]
```diff
@@ -234,7 +234,7 @@ if [ -d "${MANIFESTS}"/factory-keys ]; then
         ln -sf "${MANIFESTS}"/factory-keys/uefi conf/factory-keys/uefi
     fi
     # Link default TI K3 RoT keys if not set by the user
-    if  -d "${MANIFESTS}"/factory-keys/platform -a ! -d "conf/factory-keys/platform" ]; then
+    if [ -d "${MANIFESTS}"/factory-keys/platform -a ! -d "conf/factory-keys/platform" ]; then
```
Comment (mwasilew) [inline setup-environment-internal:237]: shellcheck suggests this isn't the best syntax:
```
In setup-environment-internal line 229:
    if [ -d "${MANIFESTS}"/factory-keys/tf-a -a ! -d "conf/factory-keys/tf-a" ]; then
                                             ^-- SC2166 (warning): Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
```

Maybe this should be updated in a separate patch.

Comment (ricardosalveti) [reply setup-environment-internal:237]: Yeah, change to the suggested one as that would be indeed better.
